bip-0341.mediawiki

```diff
@@ -124,8 +124,9 @@ If the parameters take acceptable values, the message is the concatenation of th
 * Data about this output:
 ** If ''hash_type & 3'' equals <code>SIGHASH_SINGLE</code>:
 *** ''sha_single_output'' (32): the SHA256 of the corresponding output in <code>CTxOut</code> format.
+* Extensions if ''ext_flag != 0''
 
-The total length of ''SigMsg()'' is at most ''206'' bytes<ref>'''What is the output length of ''SigMsg()''?''' The total length of ''SigMsg()'' can be computed using the following formula: ''174 - is_anyonecanpay * 49 - is_none * 32 + has_annex * 32''.</ref>. Note that this does not include the size of sub-hashes such as ''sha_prevouts'', which may be cached across signatures of the same transaction.
+The total length of ''SigMsg()'' (without extensions) is at most ''206'' bytes<ref>'''What is the output length of ''SigMsg()''?''' The total length of ''SigMsg()'' can be computed using the following formula: ''174 - is_anyonecanpay * 49 - is_none * 32 + has_annex * 32''.</ref>. Note that this does not include the size of sub-hashes such as ''sha_prevouts'', which may be cached across signatures of the same transaction.
 
```

Comment:
This is false. The total length of SigMsg is in fact at most 206 bytes. The message extension is appended to the output of SigMsg but it is not in itself part of SigMsg. In BIP 341 and 342 is clearly specified that what is hashes is not the output of SigMsg but rather 0x00 || SigMsg() || ext (if ext is present)

Reply:
I don't think what you've said and this addition are at odds. As someone who is not a core developer reading this BIP, I would be inclined to allocate exactly 206 bytes for the data to be hashed as input to my signing function. Adding the parenthetical puts me on notice that there may be cases where I need a larger buffer.

Reply:
Ok I understand your point. But the wording of the BIP is very precise, 206 bytes is the maximum total length of SigMsg(). We may clarify that what is signed is not the output SigMsg() but its output plus additional data; so the maximum total length of what is signed is not 206 bytes but 1 (version) + 206(SigMsg()) + 37(ext) = 244 bytes

Reply:
I agree with @giacomocaironi on this one. This change makes me thing "if extensions are included, SigMsg() can exceed 206 bytes."

Reply:
> The message extension is appended to the output of SigMsg but it is not in itself part of SigMsg. 

Agree. It's simpler if SigMsg remains independent of extensions. Your suggested change could be misunderstood as appending `ext` twice. First as an `Extension` inside `SigMsg` and then again due to the `||` operator.

Reply:
I've changed the commit relating to SigMsg / extensions to avoid this confusion by adding a `Taproot script path spending signature validation` section.